src/main/java/ch/uzh/ifi/hase/soprafs22/entity/User.java

```diff
@@ -94,11 +95,12 @@ public class User implements Serializable {
     @ElementCollection
     private Set<Long> likedByUsers = new TreeSet<>();
 
-    @ElementCollection
-    private Set<Long> matches = new TreeSet<>();
+    // cascade must include remove, so that the matches are deleted as well (important)
+    @ManyToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "users")
```

Comment:
@jan-kreischer To our discussion from today - From a database perspective, it makes sense that Match is the owning entity in this bidirectional relationship. Then, cascadeType.REMOVE is allowed here, since the matches must indeed be deleted.